product_configurator_refactor/models/product_attribute.py

```diff
@@ -0,0 +1,41 @@
+# -*- coding: utf-8 -*-
+# Copyright (C) 2012 - TODAY, Ursa Information Systems
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from odoo import api, fields, models, _
+from odoo.exceptions import ValidationError
+
+
+class ProductAttributeLine(models.Model):
+    _inherit = 'product.attribute.line'
+
+    @api.multi
+    def _get_attribute_values(self):
+        for line in self:
+            attribute_value_ids = line.value_idss.mapped('attrib_value_id')
+            if attribute_value_ids:
+                line.value_ids = [(6, 0, attribute_value_ids.ids)]
+
+    value_idss = fields.One2many('product.attribute.value.line',
+                                 'attribute_line_id', 'Values', copy=True)
+    value_ids = fields.Many2many(compute='_get_attribute_values',
+                                 comodel_name='product.attribute.value',
+                                 string='Attribute Values')
+    default_val = fields.Many2one('product.attribute.value',
+                                  compute='_is_default_value',
+                                  string="Default Value", store=True)
+
+    @api.one
```

Comment:
`@api.one` is deprecated: https://github.com/odoo/odoo/pull/8527

Reply:
@max3903 : I have pushed couple of commits to resolve all your above mention points.

Updated branch is: 10.0-smangukiya_refactor_product_configurator

Please find it from there. Thanks!